pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,286 @@
+	cmd.PersistentFlags().StringP(repoServerUserFlag, "u", "", "name of the user to be created for the kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerUserAccessSecretFlag, "k", "", "name of the secret having user access password and host")
+	cmd.PersistentFlags().StringP(repoAdminUserSecretFlag, "a", "", "name of the secret for the repository server admin user details")
+	cmd.PersistentFlags().StringP(repoPasswordSecretFlag, "r", "", "name of the secret containing password for the kopia repository")
+	cmd.PersistentFlags().StringP(s3LocationSecretFlag, "l", "", "name of the secret containing the s3 location details")
+	cmd.PersistentFlags().StringP(s3LocationCredsSecretFlag, "c", "", "name of the secret containing the credentials for s3")
+	cmd.PersistentFlags().StringP(prefixFlag, "p", "", "prefix to be set in kopia repository")
+	cmd.PersistentFlags().StringP(repoServerAdminUserFlag, "z", "", "kopia repository server admin user name")
+
+	_ = cmd.MarkFlagRequired(tlsSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoAdminUserSecretFlag)
+	_ = cmd.MarkFlagRequired(repoPasswordSecretFlag)
+	_ = cmd.MarkFlagRequired(s3LocationSecretFlag)
+	_ = cmd.MarkFlagRequired(s3LocationCredsSecretFlag)
+	_ = cmd.MarkFlagRequired(prefixFlag)
+	return cmd
+}
+
+func createNewRepositoryServer(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+	if len(args) > 0 {
+		return newArgsLengthError("expected 0 args. Got %#v", args)
+	}
+
+	rsParams, err := generateRepositoryServerParams(cmd)
+	if err != nil {
+		return err
+	}
+
+	repositoryServer, err := validateAndConstructRepositoryServer(rsParams)
+	if err != nil {
+		return err
+	}
+	_, crCli, _, _ := initializeClients()
+	ctx := context.Background()
+	rs, err := crCli.CrV1alpha1().RepositoryServers(defaultKanisterNamespace).Create(ctx, repositoryServer, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	} else {
+		fmt.Printf("repository-server '%s' created\n", rs.GetName())
+	}
+	return nil
+}
+
+func generateRepositoryServerParams(cmd *cobra.Command) (*repositoryServerParams, error) {
+	// Fetch values of the flags
+	tls, _ := cmd.Flags().GetString(tlsSecretFlag)
+	repoServerUser, _ := cmd.Flags().GetString(repoServerUserFlag)
+	repoServerUserAccess, _ := cmd.Flags().GetString(repoServerUserAccessSecretFlag)
+	repoAdminUser, _ := cmd.Flags().GetString(repoAdminUserSecretFlag)
+	repoPassword, _ := cmd.Flags().GetString(repoPasswordSecretFlag)
+	s3Location, _ := cmd.Flags().GetString(s3LocationSecretFlag)
+	s3LocationCreds, _ := cmd.Flags().GetString(s3LocationCredsSecretFlag)
+	prefix, _ := cmd.Flags().GetString(prefixFlag)
+	repoServerAdminUser, _ := cmd.Flags().GetString(repoServerAdminUserFlag)
+
+	var tlsNS, repoServerUserAccessNS, repoAdminUserNS, repoPasswordNS, s3LocationNS, s3LocationCredsNS string
+	if strings.Contains(tls, "/") {
```

Comment:
Would it be a good idea to just support the format `namespace/name` for these resources, instead of support both `namespace/name` and just `name`? Or there was a reason because of which we are supporting both of these formats?

Reply:
I left it just because of user convinience, if user just use `name` we'd know it's in controller namespace, and if they want to mention namespace they could mention it.

Reply:
Are you planning to raise another PR for the documentation change or going to include documentation change in this PR itself?
Can you please do that and then we can decide what should be the better way to do the things.

Reply:
I think we can include it in same PR

Reply:
Upon further discussion with @pavannd1 yesterday, I think we remove namespace/<name> and instead support only <name> as we expect the secrets to be present in kanister namespace.